zerver/lib/actions.py

```diff
@@ -4518,6 +4533,24 @@ def do_update_message(user_profile: UserProfile, message: Message,
         )
         changed_messages += messages_list
 
+        if new_stream is not None:
+            assert stream_being_edited is not None
+            UserMessage.objects.filter(
+                user_profile__in=[sub.user_profile for sub in guest_subs],
+                message__in=changed_messages,
```

Comment:
Switched these to access `_id` fields since that's clearer that we're not doing extra queries here.